packages/flame_riverpod/example/test/widget_test.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:example/main.dart';
+import 'package:flame/game.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Test equality of Flutter Text Widget and Flame Text Component',
+      (widgetTester) async {
+    await widgetTester.pumpWidget(
+      const ProviderScope(child: MyApp()),
+    );
+    await widgetTester.pump(const Duration(seconds: 3));
+
+    // Expect counter to exist
+    final flutterCounterFinder = find.byType(FlutterCountingComponent);
+    expect(flutterCounterFinder, findsOneWidget);
+    final flutterCounterTextFinder = find.descendant(
+      of: flutterCounterFinder,
+      matching: find.byType(Text),
+    );
+
+    // Expect a title 'e.g. Flutter' and the current count of the stream as
+    // seperate [Text] widget.
+    expect(flutterCounterTextFinder, findsNWidgets(2));
+
+    final flutterCounterTextWidgets =
+        widgetTester.widgetList(flutterCounterTextFinder);
+
+    // Current count of the stream as a [Text] widget
+    final flutterCounterTextWidgetOfInterest =
+        flutterCounterTextWidgets.elementAt(1);
+
+    // Expect RiverpodGameWidget to exist
+    final riverpodGameWidgetFinder = find.byType(RiverpodGameWidget);
+
+    expect(riverpodGameWidgetFinder, findsOneWidget);
+
+    // Find the game widget
+    final gameWidgetFinder = find.descendant(
+      of: riverpodGameWidgetFinder,
+      matching: find.byType(GameWidget<FlameGame>),
+    );
+    expect(gameWidgetFinder, findsOneWidget);
+
+    final gameWidget = widgetTester.widget(gameWidgetFinder);
+
+    // Initialise the game lifecycle
+    (gameWidget as GameWidget<FlameGame>).game?.lifecycle.processQueues();
+    expect(gameWidget.game?.lifecycle.hasPendingEvents, false);
+    expect(gameWidget.game?.children.isNotEmpty ?? false, true);
```

Comment:
@st-pasha can you give some pointers how to migrate this to the new lifecycle code?

Reply:
There should be no need to access the lifecycle queue directly - as long as the game was properly added to a GameWidget, and there was a `tester.pump()`, it should resolve all its lifecycle events automatically.